### PR TITLE
Move `LimitPushdown` to physical-optimizer crate

### PR DESCRIPTION
**Pull request by lewiszlw**
## Which issue does this PR close?

part of https://github.com/apache/datafusion/issues/11502.

### CODE REVIEW

File: datafusion/common/src/utils/mod.rs [hunk 1]
```diff
@@ -683,6 +683,69 @@ pub fn transpose<T>(original: Vec<Vec<T>>) -> Vec<Vec<T>> {
     }
 }
 
+/// Computes the `skip` and `fetch` parameters of a single limit that would be
```
Comment (alamb) [inline datafusion/common/src/utils/mod.rs:686]: It is nice to put this into `utils` 👍

File: datafusion/physical-optimizer/src/output_requirements.rs [hunk 2]
```diff
@@ -286,3 +286,5 @@ fn require_top_ordering_helper(
         Ok((plan, false))
     }
 }
+
+// See tests in datafusion/core/tests/physical_optimizer
```
Comment (alamb) [inline datafusion/physical-optimizer/src/output_requirements.rs:290]: ❤️

**Review by alamb — APPROVED** [review]
Thank you @lewiszlw  -- I took the liberty or mergeing this PR up from main and resolving some clippy errors so it was ready to merge

**Review by jayzhan211 — APPROVED** [review]
👍

**Comment by jayzhan211** [comment]
Thanks @lewiszlw @alamb

**Pull request merged**